src/main/deltachat/login.ts

```diff
@@ -25,207 +22,139 @@ export default class DCLoginController extends SplitOut {
+
     try {
-      await this.login(newAccountPath, credentials)
+      accountContext.configure(credentials)
     } catch (error) {
-      log.debug(
-        'Detected account creation error, deleting unfinished account',
-        newAccountPath
-      )
-      await rm(newAccountPath, { recursive: true, force: true })
+      log.debug('Detected account creation error')
+      this.dc.removeAccount(accountId)
       throw error
     }
-    const logins = await this.getLogins()
 
-    return logins.find(account => account.path === newAccountPath)
+    return accountId
+  }
+
+  async removeAccount(accountId: number) {
+    this.dc.removeAccount(accountId)
   }
 
   close() {
-    this._controller.emit('DESKTOP_CLEAR_ALL_NOTIFICATIONS')
-    if (!this._dc) return
-    this._dc.stopIO()
-    this._controller.unregisterEventHandler(this._dc)
-    this._dc.close()
-    this._controller._dc = null
+    this.controller.emit('DESKTOP_CLEAR_ALL_NOTIFICATIONS')
+    if (!this.dc) return
+    this.dc.stopIO()
+    this.controller.unregisterEventHandler(this.dc)
+    this.dc.close()
+    this.controller.dc = null
   }
 
   updateDeviceChats() {
-    this._controller.hintUpdateIfNessesary()
+    this.controller.hintUpdateIfNessesary()
 
-    this._dc.addDeviceMessage(
+    this.selectedAccountContext.addDeviceMessage(
       'changelog-version-1.20_part1_0',
       `Improved e-mail compatibility in version 1.20:
-
 📫 Read mailing lists
 📭 Read HTML-mails
 📪 Nice handling of support addresses
 ✒️💭 Read Signatures of your Contacts and use yours to express yourself, share your status or your Bio.
 📞 experimental video chat can now be used in groups
 🔧 Bug fixes and stability improvements.
-
 [Full Change log](https://github.com/deltachat/deltachat-desktop/blob/master/CHANGELOG.md#12000---2021-03-22)` as any
     )
 
-    this._dc.addDeviceMessage(
+    this.selectedAccountContext.addDeviceMessage(
       'changelog-version-1.20_part2_0',
       `With this release we are reaching out to early adopters who are willing to face some challenges and want to help bring a convenient automatically end-to-end encrypting e-mail app to all platforms. It's about time, isn't it?
 And don't worry .... Delta Chat also remains a decentralized messenger with the largest addressable user base ;)
-
 💡 We welcome your feedback and suggestions in our forum:
 https://support.delta.chat/
-
 ℹ️ You can find more information about this release in our blog post about it:
 https://delta.chat/en/2021-05-05-email-compat` as any
     )
   }
 
-  async getLogins() {
-    return await getLogins()
+  async accountInfo(accountId: number): Promise<DeltaChatAccount> {
+    const accountContext = this.dc.accountContext(accountId)
+    const selfContact = accountContext.getContact(C.DC_CONTACT_ID_SELF)
+    return {
+      accountId,
+      displayname: selfContact.getDisplayName(),
+      addr: selfContact.getAddress(),
+      path: '/dev/null',
+      size: 0,
```

Comment:
size could be interesting, path probably not so much, cause it is just core cwd plus account id

Reply:
maybe implement size, remove path